canu/generate/switch/config/config.py

```diff
@@ -638,7 +639,27 @@ def generate_switch_config(
     template_name = templates[node_shasta_name][
         "primary" if is_primary else "secondary"
     ]
+
+    def vsx_mac(switch_name):
+        is_primary, primary, secondary = switch_is_primary(switch_name)
+        primary_number = re.search(r"\d+", primary).group()
+        hex_number = format(int(primary_number), "02x")
+        if "sw-spine" in switch_name:
+            vsx_mac = "02:00:00:00:" + hex_number + ":00"
+        elif "sw-leaf" in switch_name:
+            vsx_mac = "02:01:00:00:" + hex_number + ":00"
+        elif "sw-cdu" in switch_name:
+            vsx_mac = "02:02:00:00:" + hex_number + ":00"
+        if is_valid_mac(vsx_mac):
```

Comment:
generating on hex number allows for 16 possible combinations.  is that enough for large systems?